### PR TITLE
[Silabs] add status screen

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Add a third lcd display screen displaying general information about the device.

### CODE REVIEW

File: examples/platform/silabs/display/lcd.h [hunk 1]
```diff
@@ -33,6 +33,22 @@ class SilabsLCD
...
+        InvalidScreen,
+    } Screen_e;
+
+    typedef struct dStatus
+    {
+        uint8_t nbFabric     = 0;
+        bool connected       = false;
+        char networkName[50] = { "TODO" };
+        bool advertising     = false;
+    } DisplayStatus_t;
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/display/lcd.h:50]: Do we have to maintain a status structure? or can rather we fetch them with the different apis when we have to update the LCD

File: examples/platform/silabs/display/lcd.cpp [hunk 2]
```diff
@@ -194,19 +283,14 @@ void SilabsLCD::SetQRCode(uint8_t * str, uint32_t size)
     }
 }
 
-void SilabsLCD::ShowQRCode(bool show, bool forceRefresh)
+void SilabsLCD::ShowQRCode(bool show)
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/display/lcd.cpp:286]: What is the use of the parameter `show`? It is unused and always set to true in all `ShowQRCode` calls

Comment (jepenven-silabs) [reply examples/platform/silabs/display/lcd.cpp:286]: good point, will remove it in a follow up PR.

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/platform/silabs/display/lcd.cpp [hunk 3]
```diff
@@ -149,11 +147,102 @@ void SilabsLCD::WriteDemoUI()
...
+void SilabsLCD::WriteStatus()
+{
+    uint8_t lineNb = 0;
+    char str[20];
+
+    GLIB_clear(&glibContext);
+    sprintf(str, "# Fabrics : %d", mStatus.nbFabric);
+    GLIB_drawStringOnLine(&glibContext, str, lineNb++, GLIB_ALIGN_LEFT, 0, 0, true);
+
+    if (strlen(mStatus.networkName) >= sizeof(str))
```
Comment (andy31415) [inline examples/platform/silabs/display/lcd.cpp:159]: You could use chip::Platform::CopyString here and not have extra logic for sizeof.

File: examples/platform/silabs/display/lcd.cpp [hunk 4]
```diff
@@ -149,11 +147,102 @@ void SilabsLCD::WriteDemoUI()
     }
 }
 
+void SilabsLCD::WriteStatus()
+{
+    uint8_t lineNb = 0;
+    char str[20];
+
+    GLIB_clear(&glibContext);
+    sprintf(str, "# Fabrics : %d", mStatus.nbFabric);
```
Comment (andy31415) [inline examples/platform/silabs/display/lcd.cpp:156]: maybe StringBuilder would be safer than trying to have sizes guaranteed to match.

Then this is super safe and less worry below about various strlen. Just reset the StringBuilder and use its c_str that is guaranteed to be null terminated. You may even add `AddMarkerIfOverflow`

**Review by andy31415 — APPROVED** [review]

File: examples/platform/silabs/display/lcd.cpp (hunk 4 above)
Comment (jepenven-silabs) [reply examples/platform/silabs/display/lcd.cpp:156]: Will investigate in a follow up, LCD needs rework anyway

File: examples/platform/silabs/display/lcd.cpp (hunk 3 above)
Comment (jepenven-silabs) [reply examples/platform/silabs/display/lcd.cpp:159]: Will investigate in follow up

**Pull request merged**